_includes/templates/pinnedpost.njk

```diff
@@ -0,0 +1,24 @@
+<ul class="postList">
+  {% for post in pinnedpost %}
+  <li class="postList-post">
+    <a href="{{ post.data.url }}" class="postList-title"{% if entry.data.url == url %} aria-current="page"{% endif %}>
+      {% if post.data.title %}
+        <strong>{{ post.data.title }}</strong>
+      {% else %}
+        <code>{{ post.url }}</code>
+      {% endif %}
+    </a>
+    <br>
+    <time class="postList-date" datetime="{{ post.data.date | date('MY_FORMAT') }}">
+      {{ post.data.date | date('MY_FORMAT') }}
+    </time>
+
+    {% for tag in post.data.tags %}
+    {% set page = search.page("type=tag tag='" + tag + "'") %}
+    {% if page %}
+    <a href="{{ page.data.url  }}" class="tag">{{ tag }}</a>
+    {% endif %}
+    {% endfor %}
+  </li>
+{% endfor %}
+</ul>
```

Comment:
This template is exactly the same as `postlist.njk`.
Why don't simply reuse the existing one?

Reply:
Yeah, but you can reuse the same variable to save the pinned posts:

```
{% set postlist  = search.pages("pinned=true") %}
```

Reply:
with this there aren't take pinned post, he take all post ...
```
{% set postlist  = search.pages("pinned=true") %}
{%if postlist.length == 0 %} <p>There are no pinned posts</p>{% endif %}
{% include "templates/postslist.njk" %}
```

Reply:
I mean something like this:

```
{% set postslist = search.pages("type=posts", "date=desc", 3) %}

<h1>Latest {% if postslist.length == 1 %}Post{% else %}{{ postslist.length }} Posts{% endif %}</h1>

{% include "templates/postslist.njk" %}

<hr>

{% set postlist  = search.pages("pinned=true") %}

{%if postslist.length %}
  <h2>Pinned Post:</h1>
  {% include "templates/postlist.njk" %}
  <hr>
{% endif %}
```

So pinned posts only are visible if there are any post there. If there isn't, the blog keeps looking as is now. No need to add a "no pinned posts" message because no all users will want this feature. Btw, shouldn't the pinned post be before the latest posts?

Reply:
<img width="1440" alt="Capture d’écran 2022-11-04 à 11 51 53" src="https://user-images.githubusercontent.com/97875033/199955709-7b676d4f-525c-40c0-bdff-4218477da73b.png">
If I use your code I get this

Reply:
Sorry, there's a typo. I mixed `postslist` and `postlist`.

Reply:
Its done